buildpack.md

```diff
@@ -1021,7 +1044,8 @@ name = "<dependency name>"
   cache = false
 
 [metadata]
-# buildpack-specific data
+  exec-env = ["<execution environment>"] # Optional. If not specified, applies to all execution environments
```

Comment:
Just to clarify on the platform side - if the previous image was built with a different value of `exec-env` do we do anything? Looking at the platform PR, it seems not. So do we expect buildpacks to inspect this value and re-create the layer if needed? If so we might want to call this out somewhere.